libvast/vast/system/transformer.hpp

```diff
@@ -47,10 +52,18 @@ struct transformer_state {
 /// An actor containing a transform_stream_stage, which is just a stream
 /// stream stage that applies a `transformation_engine` to every table slice.
 /// @param self The actor handle.
+/// @param transforms The set of transforms to be applied.
 transformer_actor::behavior_type
 transformer(transformer_actor::stateful_pointer<transformer_state> self,
             std::string name, std::vector<transform>&&);
 
+/// A transformer actor that is attached to a system component. This behaves
+/// identical to the regular `transformer` except for a slightly different
+/// logic during system shutdown.
+transformer_actor::behavior_type component_transformer(
```

Comment:
Shouldn't this just be an argument to the `transformer` function above rather than wrapping the function? E.g., something like

```c++
enum class source_requires_shutdown {
  yes,
  no,
};
```

Reply:
Maybe, I started that way but it seems that `spawn()` doesn't like default arguments, so we would have to have every call site understand the issue and provide the correct additional parameter. Since this is currently only relevant for the importer, this felt a bit like overkill.

Reply:
I don't think specifying the argument at all call sites is overkill—it just makes the choice more explicit. I'd prefer having it visible at all call sites, but my opinion on this is also not too strong, so proceed as you like.

Reply:
I think I'll leave it as is then, the risk that someone passes the wrong argument to me feels greater than the benefit of being explicit here.